web/client/components/style/EqualIntervalComponents/ColorRampItem.jsx

```diff
@@ -1,23 +1,36 @@
-const PropTypes = require('prop-types');
+/**
+ * Copyright 2019, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
 const React = require('react');
-const colors = require("./ExtendColorBrewer");
+const PropTypes = require('prop-types');
+const colors = require('./ExtendColorBrewer');
 const Message = require('../../I18N/Message');
-class ColorRampItem extends React.Component {
-    static propTypes = {
-        item: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
-    };
 
-    render() {
-        let ramp = this.props.item && (this.props.item.ramp || colors[this.props.item.name] && colors[this.props.item.name][5]) || [];
-        return (<div className="color-ramp-item">
-                {ramp.map(cell => <div className="color-cell" key={this.props.item && this.props.item.name + "-" + cell} style={{backgroundColor: cell}}/>)}
-                <div className="colorname-cell">
-                    {this.props.item && this.props.item.name
-                        ? <Message msgId={`global.colors.${this.props.item.name}`} msgParams={{number: ramp.length}} />
-                    : this.props.item && this.props.item.name}
-                </div>
-                </div>);
-    }
-}
+/**
+ * @name ColorRampItem
+ * @description Simple component to display color label
+ */
+const ColorRampItem = ({ item }) => {
+    let ramp = item && (item.ramp || colors[item.name] && colors[item.name][5]) || [];
+    return (
+        <div className="color-ramp-item">
+            {ramp.map(cell => <div className="color-cell" key={item && item.name + "-" + cell} style={{ backgroundColor: cell }} />)}
+            <div className="colorname-cell">
+                {item && item.name
+                    ? <Message msgId={item.name.includes('global.colors') ? item.name : `global.colors.${item.name}`} msgParams={{ number: ramp.length }} />
```

Comment:
I think this switch makes the component less generic. This way I can not customize labels for colors (that was the original way the component worked). If name of the layer is not in the global colors, you will always have an unreadable label, no way to customize it. 

Is it possible instead to add global.colors outside, when the original issue was happening, and remove the global.colors string concatenation (that should be the correct way to do the fix in https://github.com/geosolutions-it/MapStore2/pull/3900) ?

Reply:
Noted. Will push the update shortly @offtherailz